pkg/controllers/csi/gc/binaries.go

```diff
@@ -4,75 +4,41 @@ import (
 	"context"
 	"os"
 
-	"github.com/pkg/errors"
 	"github.com/spf13/afero"
 )
 
-func (gc *CSIGarbageCollector) runBinaryGarbageCollection(ctx context.Context, tenantUUID string) {
+//nolint:gosec
+func (gc *CSIGarbageCollector) runBinaryGarbageCollection(ctx context.Context) {
 	fs := &afero.Afero{Fs: gc.fs}
 
 	gcRunsMetric.Inc()
 
-	usedVersions, err := gc.db.GetUsedVersions(ctx, tenantUUID)
+	codeModules, err := gc.db.ReadCodeModules(ctx)
 	if err != nil {
-		log.Info("failed to get used versions", "error", err)
+		log.Error(err, "failed to read codemodules")
 
 		return
 	}
 
-	log.Info("got all used versions (in deprecated location)", "tenantUUID", tenantUUID, "len(usedVersions)", len(usedVersions))
-
-	storedVersions, err := gc.getStoredVersions(fs, tenantUUID)
-	if err != nil {
-		log.Info("failed to get stored versions", "error", err)
-
-		return
-	}
-
-	log.Info("got all stored versions (in deprecated location)", "tenantUUID", tenantUUID, "len(storedVersions)", len(storedVersions))
-
-	setAgentBins, err := gc.db.GetLatestVersions(ctx)
-	if err != nil {
-		log.Error(err, "failed to get the set image digests")
-	}
-
-	for _, version := range storedVersions {
-		_, isPinnedVersion := setAgentBins[version]
-
-		shouldDelete := shouldDeleteVersion(version, usedVersions) && !isPinnedVersion
-		if !shouldDelete {
-			log.Info("skipped, version should not be deleted", "version", version)
+	for _, codeModule := range codeModules {
+		orphaned, err := gc.db.IsCodeModuleOrphaned(ctx, &codeModule)
+		if err != nil {
+			log.Error(err, "failed to check if codemodule is orphaned")
 
-			continue
+			return
```

Comment:
are you sure you want to stop after first error?

Reply:
Nice catch

Reply:
Done